tutorials/25_Customizing_Agent.ipynb

```diff
@@ -0,0 +1,4851 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #8.            "transcript": agent_step.transcript,
</pre><p>Where does agent_step come from? and also then agent_step.transcript? Are resolver functions always supposed to take in these three 'query, agent and agent_step'? If yes, I would rephrase the top to:</p><ol><li>Say this exactly, a resolver function should always take in these 3 arguments</li><li>Describe what AgentStep is or link to the documentation page</li></ol><p>Further, can someone please explain why we need this? Is it optional? Or is it always needed? Are there no default resolver functions? Most agents will always have query, tool_names_withn_descriptions, transcript and memory right? </p><p>Are we only explaining this as a optional customization? If yes, perfect, but as a reader I may think it's required. So the only change I would suggest is to mention that this is something that has a default setup, but this is how you customize it :) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-tutorials/pull/223/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Reply:
As we discussed, we will add AgentStep to the documentation. I also added more information to the tutorial to clarify the points you mentioned 👍